policy/modules/admin/fapolicyd.te

```diff
@@ -0,0 +1,160 @@
+
+type fapolicyd_compiled_rules_t;
+files_type(fapolicyd_compiled_rules_t)
+
+type fapolicyd_log_t;
+logging_log_file(fapolicyd_log_t)
+
+type fapolicyd_runtime_t;
+files_runtime_file(fapolicyd_runtime_t)
+
+type fagenrules_tmp_t;
+files_tmp_file(fagenrules_tmp_t)
+
+type fapolicyd_unit_file_t;
+init_unit_file(fapolicyd_unit_file_t)
+
+type fapolicyd_var_lib_t;
+files_type(fapolicyd_var_lib_t)
+
+########################################
+#
+# fapolicyd (daemon) local policy
+#
+
+allow fapolicyd_t self:capability { audit_write chown dac_override setgid setuid sys_admin sys_nice sys_ptrace };
+allow fapolicyd_t self:process { setcap setsched };
+
+manage_fifo_files_pattern(fapolicyd_t, fapolicyd_runtime_t, fapolicyd_runtime_t)
+manage_files_pattern(fapolicyd_t, fapolicyd_runtime_t, fapolicyd_runtime_t)
+mmap_manage_files_pattern(fapolicyd_t,fapolicyd_var_lib_t,fapolicyd_var_lib_t)
+
+mmap_read_files_pattern(fapolicyd_t,fapolicyd_config_t,fapolicyd_config_t)
+files_runtime_filetrans(fapolicyd_t, fapolicyd_runtime_t, { file fifo_file })
+
+read_files_pattern(fapolicyd_t, fapolicyd_config_t, fapolicyd_compiled_rules_t)
+
+fapolicyd_read_config_files(fapolicyd_t)
+
+domain_read_all_entry_files(fapolicyd_t)
+domain_read_all_domains_state(fapolicyd_t)
+
+files_read_all_files(fapolicyd_t)
+files_read_all_symlinks(fapolicyd_t)
+files_map_usr_files(fapolicyd_t)
+files_watch_all_mountpoints(fapolicyd_t)
+
+fs_getattr_xattr_fs(fapolicyd_t)
+
+kernel_getattr_proc(fapolicyd_t)
+kernel_read_kernel_sysctls(fapolicyd_t)
+kernel_read_crypto_sysctls(fapolicyd_t)
+
+logging_log_filetrans(fapolicyd_t, fapolicyd_log_t, file)
+logging_send_syslog_msg(fapolicyd_t)
+
+optional_policy(`
+	rpm_manage_db(fapolicyd_t)
+')
+
+########################################
+#
+# fapolicygen local policy
+#
+
+allow fagenrules_t self:capability fsetid;
+allow fagenrules_t self:fifo_file rw_inherited_fifo_file_perms;
+
+read_files_pattern(fagenrules_t, fapolicyd_config_t, fapolicyd_compiled_rules_t)
+allow fagenrules_t fapolicyd_compiled_rules_t:file setattr;
+
+manage_files_pattern(fagenrules_t, fapolicyd_var_lib_t, fapolicyd_var_lib_t)
+
+# script calls restorecon on /etc/fapolicy/compiled.rules
+allow fagenrules_t fapolicyd_compiled_rules_t:file relabelto;
+allow fagenrules_t fapolicyd_config_t:file relabelfrom;
+manage_files_pattern(fagenrules_t,fapolicyd_config_t,fapolicyd_config_t)
+
+# fagenpolicy finds running fapolicyd and sighup after generating rules
+allow fagenrules_t fapolicyd_t:process signal;
+ps_process_pattern(fagenrules_t, fapolicyd_t)
+
+fapolicyd_read_config_files(fagenrules_t)
+
+corecmd_exec_bin(fagenrules_t)
+corecmd_exec_shell(fagenrules_t)
+
+# fagenpolicy uses 'pidof' to find running fapolicyd
+domain_dontaudit_read_all_domains_state(fagenrules_t)
+
+files_list_runtime(fagenrules_t)
+files_read_etc_files(fagenrules_t)
+
+files_tmp_filetrans(fagenrules_t, fagenrules_tmp_t, { dir file })
+manage_files_pattern(fagenrules_t, fagenrules_tmp_t, fagenrules_tmp_t)
```

Comment:
Needs line ordering style cleanup.

Reply:
I think I got this correct, again, please let me know if there is anything else.